analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.FlowCaptureOperation.cs

```diff
@@ -0,0 +1,354 @@
+    {
+        /*       Block 1
+         *       #0=s1
+         *         |
+         *       Block 2
+         *   #1=(s1="overwrite")
+         *     #1 is null
+         *        /   \
+         *       T     F
+         *       /     \
+         *    Block3   Block4
+         *    #2=#1   #2="value"
+         *      \         /
+         *       \       /
+         *        Block5
+         *        s1=#2
+         */
+        const string code = """s1 = (s1 = "overwrite")  ?? "value";""";
+        var context = CreateContextCS(code, additionalParameters: "string s1");
+        context.ValidateEntry(LiveIn("s1"), LiveOut("s1"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("s1"));
+        context.Validate(context.Cfg.Blocks[2], null);
+        context.Validate(context.Cfg.Blocks[3], null, LiveOut("s1"));
+        context.Validate(context.Cfg.Blocks[4], null, LiveOut("s1"));
+        context.Validate(context.Cfg.Blocks[5], null, LiveIn("s1"));
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_SwitchStatement()
+    {
+        const string code = """
+            string result = i switch
+            {
+                0 =>  stringParam,
+                1 => "Something",
+                _ => "Everything"
+            };
+            """;
+        var context = CreateContextCS(code, additionalParameters: "int i, string stringParam");
+        context.ValidateEntry(LiveIn("i", "stringParam"), LiveOut("i", "stringParam"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("i", "stringParam"), LiveOut("i", "stringParam")); // if i == 0
+        context.Validate(context.Cfg.Blocks[2], null, LiveIn("stringParam"), LiveOut("stringParam"));           // #0 = stringParam
+        context.Validate(context.Cfg.Blocks[3], null, LiveIn("i", "stringParam"), LiveOut("i", "stringParam")); // if i == 1
+        context.Validate(context.Cfg.Blocks[4], null, LiveIn("stringParam"), LiveOut("stringParam"));           // #0 = "Something"
+        context.Validate(context.Cfg.Blocks[5], null, LiveIn("i", "stringParam"), LiveOut("stringParam"));      // if i != 0 && i != 1
+        context.Validate(context.Cfg.Blocks[6], null, LiveIn("stringParam"), LiveOut("stringParam"));           // #0 = "Everything"
+        context.Validate(context.Cfg.Blocks[7], null);                                                          // to no destination block
+        context.Validate(context.Cfg.Blocks[8], null, LiveIn("stringParam"));                                   // result = #0
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_ForEachCompundAssignment()
+    {
+        const string code = """
+            int sum = 0;
+            foreach (var i in list)
+            {
+                sum += i;
+            }
+            """;
+        var context = CreateContextCS(code, additionalParameters: "System.Collections.Generic.List<int> list");
```

Comment:
Does this produce something different?
```suggestion
        var context = CreateContextCS(code, additionalParameters: "int[] array");
```